util/expensivequery/expensivequery.go

```diff
@@ -73,12 +82,213 @@ func (eqh *Handle) Run() {
 				}
 			}
 			threshold = atomic.LoadUint64(&variable.ExpensiveQueryTimeThreshold)
+			eqh.oomKillerAlert()
 		case <-eqh.exitCh:
 			return
 		}
 	}
 }
 
+type oomRecord struct {
+	oomRecordErr      error
+	oomRecordStatus   status // 0 uninitialized, 1 instance memory usage, 2 system memory usage, 3 close the alert.
+	serverMemoryQuota uint64
+	lastOOMtime       time.Time
+
+	tmpDir              string
+	lastLogFileName     []string
+	lastProfileFileName []string
+}
+
+type status uint32
+
+const (
+	uninitialized status = iota
```

Comment:
1. We do not need `uninitialized` and `closeAlert`.
We can init in `Handle.Run` before the for loop.

2. We can use a bool value to check `useInstanceMemory ` or `useSystemMemory `

Reply:
addressed.